skafka/src/main/scala/com/evolutiongaming/skafka/Blocking.scala

```diff
@@ -1,14 +1,28 @@
 package com.evolutiongaming.skafka
 
+import cats.effect.{Async, ContextShift}
+import com.evolutiongaming.catshelper.FromFuture
+
 import scala.concurrent.{ExecutionContext, Future}
 
 trait Blocking[F[_]] {
-  def apply[T](f: => T): F[T]
+  def apply[A](thunk: => A): F[A]
+
+  def future[A](f: => Future[A]): F[A]
 }
 
 object Blocking {
+  def apply[F[_] : Async : ContextShift : FromFuture](blockingEC: ExecutionContext): Blocking[F] =
+    new Blocking[F] {
+      def blocking[A](f: F[A]): F[A] = ContextShift[F].evalOn(blockingEC)(f)
 
-  def future(ec: ExecutionContext): Blocking[Future] = new Blocking[Future] {
-    def apply[T](f: => T): Future[T] = Future(f)(ec)
-  }
-}
+      override def apply[A](thunk: => A): F[A] = blocking {
+        Async[F].delay(thunk)
```

Comment:
Please change everywhere Async -> Sync

Reply:
Indeed, we don't need `Async` since `FromFuture` takes care of it for us. Done!